eeps/eep-0066.md

```diff
@@ -0,0 +1,432 @@
+which is a name composed of a sequence of characters that are allowed
+as the second or later characters in a variable or an atom.
+In short ISO [Latin-1][] letters, digits, «`_`» and «`@`».
+The [Sigil Type][] may be empty.
+
+The Sigil Type defines how the [Sigil][] syntactic sugar
+shall be interpreted.  The suggested Sigil Types are:
+
+* «»: the Vanilla [Sigil][].
+
+  Creates an Erlang `unicode:unicode_binary()`.
+  It is a string represented as a UTF-8 encoded binary,
+  equivalent to applying `unicode:characters_to_binary/1`
+  on the [String Content][].  The [String Delimiters][]
+  and escape characters work as for regular strings,
+  triple-quoted strings, or quoted atoms in Erlang.
+
+  So «`~"abc\d"`» is equivalent to «`<<"abc\d"/utf8>>`», and
+  «`~'abc"d'`» is equivalent to «`<<"abc\"d"/utf8>>`».
+
+  «`~"» would work as «`~s"`» and «`~"""`» would work
+  as «`~S"""`» below, regarding escape characters.
+
+  A simple way to create strings as UTF-8 binaries is supposedly
+  the first and most desired missing string feature in Erlang.
+  This sigil does just that and has no other features.
+
+* «`s`»: [string in Elixir][4].
+
+  Creates an Erlang `unicode:unicode_binary()`, handling
+  escape characters in the string content.  How other features
+  like string interpolation would work is still an open question.
+  
+  Escape characters and other features are the same regardless
+  of which [String Delimiters][] that are used.
+
+* «`S`»: [string in Elixir][4], verbatim.
+
+  Creates an Erlang `unicode:unicode_binary()`, with verbatim
+  string content in that only the [end delimiter][] character
+  can be escaped with a «`\`» character.  How other features
+  like string interpolation would work is still an open question.
+
+  Escape characters and other features are the same regardless
+  of which [String Delimiters][] that are used, except that
+  between triple-quote delimiters according to [EEP 64][]
+  there is no end delimiter character to escape.
+
+* «`r`»: regular expression.
+
+  Creates a term `{RE::unicode:charlist(),Flags::[unicode:latin1_char()]}`
+  that is an uncompiled regular expression with compile flags,
+  suitable for functions in the `re` module.  The `RE` element is
+  the [String Content][], and the `Flags` element is the [Sigil Suffix][].
+
+  See the [Regular Expressions][] section about the reasoning
+  behind this proposed term type.
+
+  Within the [String Content][], character escape sequences are handled
+  according to the regular expression rules, and in addition to them
+  the [end delimiter][] character can be escaped with a «`\`» character.
+  Between triple-quote delimiters according to [EEP 64][]
+  there is no end delimiter character to escape.
+
+### String Delimiters
+
+Immediately following the [Sigil Type][] is the string start delimiter.
+A specific start delimiter character has a corresponding
+end delimiter character.
+
+These are the start delimiter characters and they have themselves
```

Comment:
I understand correctly that we do not have plans for supporting delimiters like Elixir, which could be:
```
~r/hello/
~r|hello|
~r"hello"
~r'hello'
~r(hello)
~r[hello]
~r{hello}
~r<hello>
```
https://elixir-lang.org/getting-started/sigils.html

If so, I agree that it is better to start small and do incremental changes / additions if needed in the future.

Reply:
It was in the EEP up to just recently.  It is a possible future augmentation.  Therefore the EEP still talks about start and end delimiter characters instead of just delimiter character.

By reducing to just two delimiters I also made the choice that the sigil name sets escape rules, not the chosen delimiters.  Like in Elixir.

Reply:
@josevalim said that he hasn't seen much use for other delimiters than `"`.  We have `'` for quoted atoms, so I kept them too.

Reply:
> @josevalim said that he hasn't seen much use for other delimiters than `"`. We have `'` for quoted atoms, so I kept them too.

`\` does seem to be used for regular expressions. But strangely enough for all. Grepping in the `elixir` repository:

```
% git grep "~r/" | wc
     414    3241   44627
% git grep "~r""" | wc 
    1028    9004  117317
```

(I am surprised that `~r/` is not used for the majority of regexes.)